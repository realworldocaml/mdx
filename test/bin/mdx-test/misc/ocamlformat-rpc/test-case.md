Toplevel blocks are formatted when setting option `--format-code`:

```ocaml
# module X = struct let x = 3 end
# let x = 2 in x + 1
# let f = function
  | (_, (_, _)) when (true || false) ->
  assert true;
  ignore @@ (fun x -> (ignore @@ (fun _ -> [2; 3; 3; 4; 5])))
  | _ -> ()
# let ( || ) x y = match x with true -> y | false -> false
```

Invalid blocks are not formatted:

```ocaml
# let x = 2 in
```

Regular OCaml blocks are not formatted:

```ocaml
let _ = let x = 2 in let y = 3 in x + y

module X = struct let x = 3 end
```
