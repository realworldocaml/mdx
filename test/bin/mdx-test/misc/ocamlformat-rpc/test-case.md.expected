Toplevel blocks are formatted when setting option `--format-code`:

```ocaml
# module X = struct
    let x = 3
  end;;
module X : sig val x : int end
# let x = 2 in
  x + 1;;
- : int = 3
```

Invalid blocks are not formatted:

```ocaml
# let x = 2 in
Line 2, characters 1-3:
Error: Syntax error
```

Regular OCaml blocks are not formatted:

```ocaml
let _ = let x = 2 in let y = 3 in x + y

module X = struct let x = 3 end
```
